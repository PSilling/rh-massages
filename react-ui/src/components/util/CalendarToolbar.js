// react imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// util imports
import _t from '../../util/Translations';

/**
 * Navigation toolbar for event calendar.
 */
class CalendarToolbar extends Component {

  render() {
    return (
      <div>
        <div className="row">
          <div className="col-md-1">
            <button type="button" className="btn btn-default" title={ _t.translate("Previous") }
              onClick={this.props.leftAction}>
              <span className="glyphicon glyphicon-chevron-left"></span>
            </button>
          </div>

          <div className="col-md-10 text-center">
            <div className="btn-group">
              <button type="button" className={"btn btn-default" + (!this.props.monthActive ? " active" : "")}
                onClick={() => this.props.onViewChange("work_week")}>
                { _t.translate('Week') }
              </button>

              <button type="button" className={"btn btn-default" + (this.props.monthActive ? " active" : "")}
                onClick={() => this.props.onViewChange("month")}>
                { _t.translate('Month') }
              </button>
            </div>
          </div>

          <div className="col-md-1 text-right">
            <button type="button" className="btn btn-default" title={ _t.translate("Next") }
              onClick={this.props.rightAction}>
              <span className="glyphicon glyphicon-chevron-right"></span>
            </button>
          </div>

        </div>
        <div className="text-center" style={{'marginBottom': '5px', marginTop: '5px'}}>
          <strong>
            {this.props.month}
          </strong>
        </div>
      </div>
    );
  }
}

CalendarToolbar.propTypes = {
  /** current selected month */
  month: PropTypes.string,
  /** whether month is the currently selected view */
  monthActive: PropTypes.bool.isRequired,
  /** function called on left chevron button click */
  leftAction: PropTypes.func.isRequired,
  /** function called on right chevron button click */
  rightAction: PropTypes.func.isRequired,
  /** function called on view change */
  onViewChange: PropTypes.func.isRequired
};

export default CalendarToolbar
