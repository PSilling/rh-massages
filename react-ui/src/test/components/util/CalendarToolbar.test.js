// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import CalendarToolbar from '../../../components/util/CalendarToolbar';

test('renders content with correct props', () => {
  const testLeftFunction = jest.fn(),
        testRightFunction = jest.fn(),
        testViewFunction = jest.fn((string) => { return string; }),
        testRenderer = TestRenderer.create(
          <CalendarToolbar
            month="test"
            monthActive={true}
            leftDisabled={false}
            rightDisabled={true}
            leftAction={testLeftFunction}
            rightAction={testRightFunction}
            onViewChange={testViewFunction}
          />
        ),
        testInstance = testRenderer.root;

  let buttons = testInstance.findAllByType('button'),
      iconLeft = testInstance.findByProps({ className: "glyphicon glyphicon-chevron-left" }),
      iconRight = testInstance.findByProps({ className: "glyphicon glyphicon-chevron-right" }),
      month = testInstance.findByType('strong'),
      treeJSON = testRenderer.toJSON();

  expect(buttons[0].props.onClick).toBe(testLeftFunction);
  expect(buttons[0].props.disabled).toBe(false);
  expect(buttons[1].props.onClick()).toBe("work_week");
  expect(buttons[1].props.className).toBe("btn btn-default");
  expect(buttons[2].props.onClick()).toBe("month");
  expect(buttons[2].props.className).toBe("btn btn-default active");
  expect(buttons[3].props.onClick).toBe(testRightFunction);
  expect(buttons[3].props.disabled).toBe(true);
  expect(month.props.children).toEqual("test");
  expect(treeJSON).toMatchSnapshot();
});
