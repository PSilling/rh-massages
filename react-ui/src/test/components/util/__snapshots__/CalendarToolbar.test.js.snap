// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CalendarToolbar
    leftAction={[MockFunction]}
    leftDisabled={false}
    month="test"
    monthActive={true}
    onViewChange={
      [MockFunction] {
        "calls": Array [
          Array [
            "work_week",
          ],
          Array [
            "month",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": "work_week",
          },
          Object {
            "isThrow": false,
            "value": "month",
          },
        ],
      }
    }
    rightAction={[MockFunction]}
    rightDisabled={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row
          tag="div"
        >
          <Col
            className="text-center"
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <strong>
              test
            </strong>
          </Col>
        </Row>,
        <Row
          className="mb-3"
          tag="div"
        >
          <Col
            md="1"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <TooltipIconButton
              disabled={false}
              icon="chevron-circle-left"
              onClick={[MockFunction]}
              size="md"
              style={
                Object {
                  "border": "0px solid transparent",
                }
              }
              tooltip="Předchozí měsíc"
            />
          </Col>
          <Col
            className="text-center"
            md="10"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <ButtonGroup
              role="group"
              tag="div"
            >
              <Button
                active={false}
                className="ml-2"
                color="secondary"
                id="Tooltip1"
                onClick={[Function]}
                outline={true}
                tag="button"
              >
                Týden
              </Button>
              <Button
                active={true}
                color="secondary"
                id="Tooltip2"
                onClick={[Function]}
                outline={true}
                tag="button"
              >
                Měsíc
              </Button>
            </ButtonGroup>
            <TooltipGroup
              labels={
                Array [
                  "Zobrazit po týdnech",
                  "Zobrazit po měsících",
                ]
              }
              targets={
                Array [
                  "Tooltip1",
                  "Tooltip2",
                ]
              }
            />
          </Col>
          <Col
            className="text-right"
            md="1"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <TooltipIconButton
              disabled={true}
              icon="chevron-circle-right"
              onClick={[MockFunction]}
              size="md"
              style={
                Object {
                  "border": "0px solid transparent",
                }
              }
              tooltip="Další měsíc"
            />
          </Col>
        </Row>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            className="text-center"
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <strong>
              test
            </strong>
          </Col>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <strong>
              test
            </strong>,
            "className": "text-center",
            "md": "12",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test",
            },
            "ref": null,
            "rendered": "test",
            "type": "strong",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              md="1"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <TooltipIconButton
                disabled={false}
                icon="chevron-circle-left"
                onClick={[MockFunction]}
                size="md"
                style={
                  Object {
                    "border": "0px solid transparent",
                  }
                }
                tooltip="Předchozí měsíc"
              />
            </Col>,
            <Col
              className="text-center"
              md="10"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <ButtonGroup
                role="group"
                tag="div"
              >
                <Button
                  active={false}
                  className="ml-2"
                  color="secondary"
                  id="Tooltip1"
                  onClick={[Function]}
                  outline={true}
                  tag="button"
                >
                  Týden
                </Button>
                <Button
                  active={true}
                  color="secondary"
                  id="Tooltip2"
                  onClick={[Function]}
                  outline={true}
                  tag="button"
                >
                  Měsíc
                </Button>
              </ButtonGroup>
              <TooltipGroup
                labels={
                  Array [
                    "Zobrazit po týdnech",
                    "Zobrazit po měsících",
                  ]
                }
                targets={
                  Array [
                    "Tooltip1",
                    "Tooltip2",
                  ]
                }
              />
            </Col>,
            <Col
              className="text-right"
              md="1"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <TooltipIconButton
                disabled={true}
                icon="chevron-circle-right"
                onClick={[MockFunction]}
                size="md"
                style={
                  Object {
                    "border": "0px solid transparent",
                  }
                }
                tooltip="Další měsíc"
              />
            </Col>,
          ],
          "className": "mb-3",
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <TooltipIconButton
                disabled={false}
                icon="chevron-circle-left"
                onClick={[MockFunction]}
                size="md"
                style={
                  Object {
                    "border": "0px solid transparent",
                  }
                }
                tooltip="Předchozí měsíc"
              />,
              "md": "1",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "icon": "chevron-circle-left",
                "onClick": [MockFunction],
                "size": "md",
                "style": Object {
                  "border": "0px solid transparent",
                },
                "tooltip": "Předchozí měsíc",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ButtonGroup
                  role="group"
                  tag="div"
                >
                  <Button
                    active={false}
                    className="ml-2"
                    color="secondary"
                    id="Tooltip1"
                    onClick={[Function]}
                    outline={true}
                    tag="button"
                  >
                    Týden
                  </Button>
                  <Button
                    active={true}
                    color="secondary"
                    id="Tooltip2"
                    onClick={[Function]}
                    outline={true}
                    tag="button"
                  >
                    Měsíc
                  </Button>
                </ButtonGroup>,
                <TooltipGroup
                  labels={
                    Array [
                      "Zobrazit po týdnech",
                      "Zobrazit po měsících",
                    ]
                  }
                  targets={
                    Array [
                      "Tooltip1",
                      "Tooltip2",
                    ]
                  }
                />,
              ],
              "className": "text-center",
              "md": "10",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <Button
                      active={false}
                      className="ml-2"
                      color="secondary"
                      id="Tooltip1"
                      onClick={[Function]}
                      outline={true}
                      tag="button"
                    >
                      Týden
                    </Button>,
                    <Button
                      active={true}
                      color="secondary"
                      id="Tooltip2"
                      onClick={[Function]}
                      outline={true}
                      tag="button"
                    >
                      Měsíc
                    </Button>,
                  ],
                  "role": "group",
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": false,
                      "children": "Týden",
                      "className": "ml-2",
                      "color": "secondary",
                      "id": "Tooltip1",
                      "onClick": [Function],
                      "outline": true,
                      "tag": "button",
                    },
                    "ref": null,
                    "rendered": "Týden",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": true,
                      "children": "Měsíc",
                      "color": "secondary",
                      "id": "Tooltip2",
                      "onClick": [Function],
                      "outline": true,
                      "tag": "button",
                    },
                    "ref": null,
                    "rendered": "Měsíc",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "labels": Array [
                    "Zobrazit po týdnech",
                    "Zobrazit po měsících",
                  ],
                  "targets": Array [
                    "Tooltip1",
                    "Tooltip2",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <TooltipIconButton
                disabled={true}
                icon="chevron-circle-right"
                onClick={[MockFunction]}
                size="md"
                style={
                  Object {
                    "border": "0px solid transparent",
                  }
                }
                tooltip="Další měsíc"
              />,
              "className": "text-right",
              "md": "1",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": true,
                "icon": "chevron-circle-right",
                "onClick": [MockFunction],
                "size": "md",
                "style": Object {
                  "border": "0px solid transparent",
                },
                "tooltip": "Další měsíc",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row
            tag="div"
          >
            <Col
              className="text-center"
              md="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <strong>
                test
              </strong>
            </Col>
          </Row>,
          <Row
            className="mb-3"
            tag="div"
          >
            <Col
              md="1"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <TooltipIconButton
                disabled={false}
                icon="chevron-circle-left"
                onClick={[MockFunction]}
                size="md"
                style={
                  Object {
                    "border": "0px solid transparent",
                  }
                }
                tooltip="Předchozí měsíc"
              />
            </Col>
            <Col
              className="text-center"
              md="10"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <ButtonGroup
                role="group"
                tag="div"
              >
                <Button
                  active={false}
                  className="ml-2"
                  color="secondary"
                  id="Tooltip1"
                  onClick={[Function]}
                  outline={true}
                  tag="button"
                >
                  Týden
                </Button>
                <Button
                  active={true}
                  color="secondary"
                  id="Tooltip2"
                  onClick={[Function]}
                  outline={true}
                  tag="button"
                >
                  Měsíc
                </Button>
              </ButtonGroup>
              <TooltipGroup
                labels={
                  Array [
                    "Zobrazit po týdnech",
                    "Zobrazit po měsících",
                  ]
                }
                targets={
                  Array [
                    "Tooltip1",
                    "Tooltip2",
                  ]
                }
              />
            </Col>
            <Col
              className="text-right"
              md="1"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <TooltipIconButton
                disabled={true}
                icon="chevron-circle-right"
                onClick={[MockFunction]}
                size="md"
                style={
                  Object {
                    "border": "0px solid transparent",
                  }
                }
                tooltip="Další měsíc"
              />
            </Col>
          </Row>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Col
              className="text-center"
              md="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <strong>
                test
              </strong>
            </Col>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <strong>
                test
              </strong>,
              "className": "text-center",
              "md": "12",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": "strong",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Col
                md="1"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <TooltipIconButton
                  disabled={false}
                  icon="chevron-circle-left"
                  onClick={[MockFunction]}
                  size="md"
                  style={
                    Object {
                      "border": "0px solid transparent",
                    }
                  }
                  tooltip="Předchozí měsíc"
                />
              </Col>,
              <Col
                className="text-center"
                md="10"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <ButtonGroup
                  role="group"
                  tag="div"
                >
                  <Button
                    active={false}
                    className="ml-2"
                    color="secondary"
                    id="Tooltip1"
                    onClick={[Function]}
                    outline={true}
                    tag="button"
                  >
                    Týden
                  </Button>
                  <Button
                    active={true}
                    color="secondary"
                    id="Tooltip2"
                    onClick={[Function]}
                    outline={true}
                    tag="button"
                  >
                    Měsíc
                  </Button>
                </ButtonGroup>
                <TooltipGroup
                  labels={
                    Array [
                      "Zobrazit po týdnech",
                      "Zobrazit po měsících",
                    ]
                  }
                  targets={
                    Array [
                      "Tooltip1",
                      "Tooltip2",
                    ]
                  }
                />
              </Col>,
              <Col
                className="text-right"
                md="1"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <TooltipIconButton
                  disabled={true}
                  icon="chevron-circle-right"
                  onClick={[MockFunction]}
                  size="md"
                  style={
                    Object {
                      "border": "0px solid transparent",
                    }
                  }
                  tooltip="Další měsíc"
                />
              </Col>,
            ],
            "className": "mb-3",
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <TooltipIconButton
                  disabled={false}
                  icon="chevron-circle-left"
                  onClick={[MockFunction]}
                  size="md"
                  style={
                    Object {
                      "border": "0px solid transparent",
                    }
                  }
                  tooltip="Předchozí měsíc"
                />,
                "md": "1",
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "icon": "chevron-circle-left",
                  "onClick": [MockFunction],
                  "size": "md",
                  "style": Object {
                    "border": "0px solid transparent",
                  },
                  "tooltip": "Předchozí měsíc",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ButtonGroup
                    role="group"
                    tag="div"
                  >
                    <Button
                      active={false}
                      className="ml-2"
                      color="secondary"
                      id="Tooltip1"
                      onClick={[Function]}
                      outline={true}
                      tag="button"
                    >
                      Týden
                    </Button>
                    <Button
                      active={true}
                      color="secondary"
                      id="Tooltip2"
                      onClick={[Function]}
                      outline={true}
                      tag="button"
                    >
                      Měsíc
                    </Button>
                  </ButtonGroup>,
                  <TooltipGroup
                    labels={
                      Array [
                        "Zobrazit po týdnech",
                        "Zobrazit po měsících",
                      ]
                    }
                    targets={
                      Array [
                        "Tooltip1",
                        "Tooltip2",
                      ]
                    }
                  />,
                ],
                "className": "text-center",
                "md": "10",
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <Button
                        active={false}
                        className="ml-2"
                        color="secondary"
                        id="Tooltip1"
                        onClick={[Function]}
                        outline={true}
                        tag="button"
                      >
                        Týden
                      </Button>,
                      <Button
                        active={true}
                        color="secondary"
                        id="Tooltip2"
                        onClick={[Function]}
                        outline={true}
                        tag="button"
                      >
                        Měsíc
                      </Button>,
                    ],
                    "role": "group",
                    "tag": "div",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "active": false,
                        "children": "Týden",
                        "className": "ml-2",
                        "color": "secondary",
                        "id": "Tooltip1",
                        "onClick": [Function],
                        "outline": true,
                        "tag": "button",
                      },
                      "ref": null,
                      "rendered": "Týden",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "active": true,
                        "children": "Měsíc",
                        "color": "secondary",
                        "id": "Tooltip2",
                        "onClick": [Function],
                        "outline": true,
                        "tag": "button",
                      },
                      "ref": null,
                      "rendered": "Měsíc",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "labels": Array [
                      "Zobrazit po týdnech",
                      "Zobrazit po měsících",
                    ],
                    "targets": Array [
                      "Tooltip1",
                      "Tooltip2",
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <TooltipIconButton
                  disabled={true}
                  icon="chevron-circle-right"
                  onClick={[MockFunction]}
                  size="md"
                  style={
                    Object {
                      "border": "0px solid transparent",
                    }
                  }
                  tooltip="Další měsíc"
                />,
                "className": "text-right",
                "md": "1",
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": true,
                  "icon": "chevron-circle-right",
                  "onClick": [MockFunction],
                  "size": "md",
                  "style": Object {
                    "border": "0px solid transparent",
                  },
                  "tooltip": "Další měsíc",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
